scripts/docker/build_msgs_jar.Dockerfile

```diff
@@ -21,18 +21,23 @@
 
 ARG REMOTE=astrobee
 ARG REPO=astrobee
-FROM ${REMOTE}/${REPO}:msgs-ubuntu16.04
-
-RUN apt-get update && apt-get install -y \
-  unzip \
-  libc6-dev-i386 \
-  lib32z1 \
-  python-wstool \
-  openjdk-8-jdk \
-  ros-kinetic-rosjava \
-  && rm -rf /var/lib/apt/lists/*
-
-# Compile msg jar files, genjava_message_artifacts only works with bash
+FROM ${REMOTE}/${REPO}:msgs-ubuntu20.04
+
+COPY scripts/setup/debians/rosjava /src/msgs/src/scripts
+
+# install dependencies for rosjava build script
+RUN apt-get update && apt-get install -y devscripts equivs
+
+# make sure there is a python binary
+RUN update-alternatives --install /usr/bin/python python /usr/bin/python3 1
+
+# build rosjava debians
+RUN cd /src/msgs/src/scripts && /bin/bash build_rosjava_debians.sh --install-with-deps
+
+# remove to save disk space
+RUN rm -rf /var/lib/apt/lists/*
```

Comment:
Not a big deal, but a point of interest is that if you want to optimize the size of a docker image by eliminating temp files, it's important to combine the temp file generating action with the cleanup action in the same `RUN` step, like `RUN make files && rm files`. This is because `docker build` creates a new image layer for each `RUN` command, so if you separate the actions in different steps, there will still be a layer containing the temp files (and using disk space) even if the temp files are not present in the top layer.

(Or you can take a different approach and squash all the layers into one at the end before export, in which case this becomes a non-issue.)

Reply:
Good point, I think I'll move the `rm` to the previous RUN only, since the build_rosjava_debians.sh needs the apt lists from the previous command and I don't think combining them will make much of a difference with size.